applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -279,184 +290,176 @@ static int test_tone_button_press(void)
+			}
+			/* Starts/pauses the audio stream */
+			ret = le_audio_play_pause();
+			if (ret) {
+				LOG_WRN("Could not play/pause");
+			}
 
-	case BUTTON_VOLUME_UP:
-		ret = le_audio_volume_up();
-		if (ret) {
-			LOG_WRN("Failed to increase volume");
-		}
+			break;
 
-		break;
+		case BUTTON_VOLUME_UP:
+			ret = le_audio_volume_up();
+			if (ret) {
+				LOG_WRN("Failed to increase volume");
+			}
 
-	case BUTTON_VOLUME_DOWN:
-		ret = le_audio_volume_down();
-		if (ret) {
-			LOG_WRN("Failed to decrease volume");
-		}
+			break;
 
-		break;
+		case BUTTON_VOLUME_DOWN:
+			ret = le_audio_volume_down();
+			if (ret) {
+				LOG_WRN("Failed to decrease volume");
+			}
 
-	case BUTTON_4:
-#if (CONFIG_AUDIO_TEST_TONE)
-		if (IS_ENABLED(CONFIG_WALKIE_TALKIE_DEMO)) {
-			LOG_DBG("Test tone is disabled in walkie-talkie mode");
 			break;
-		}
 
-		ret = test_tone_button_press();
+		case BUTTON_4:
+#if (CONFIG_AUDIO_TEST_TONE)
+			if (IS_ENABLED(CONFIG_WALKIE_TALKIE_DEMO)) {
+				LOG_DBG("Test tone is disabled in walkie-talkie mode");
+				break;
+			}
+
+			ret = test_tone_button_press();
 #else
-		ret = le_audio_user_defined_button_press(LE_AUDIO_USER_DEFINED_ACTION_1);
+			ret = le_audio_user_defined_button_press(LE_AUDIO_USER_DEFINED_ACTION_1);
 #endif /*CONFIG_AUDIO_TEST_TONE*/
 
-		if (ret) {
-			LOG_WRN("Failed button 4 press, ret: %d", ret);
-		}
+			if (ret) {
+				LOG_WRN("Failed button 4 press, ret: %d", ret);
+			}
 
-		break;
+			break;
 
-	case BUTTON_5:
+		case BUTTON_5:
 #if (CONFIG_AUDIO_MUTE)
-		ret = le_audio_volume_mute();
-		if (ret) {
-			LOG_WRN("Failed to mute volume");
-		}
+			ret = le_audio_volume_mute();
+			if (ret) {
+				LOG_WRN("Failed to mute volume");
+			}
 #else
-		ret = le_audio_user_defined_button_press(LE_AUDIO_USER_DEFINED_ACTION_2);
-		if (ret) {
-			LOG_WRN("User defined button 5 action failed, ret: %d", ret);
-		}
+			ret = le_audio_user_defined_button_press(LE_AUDIO_USER_DEFINED_ACTION_2);
+			if (ret) {
+				LOG_WRN("User defined button 5 action failed, ret: %d", ret);
+			}
 #endif
-		break;
+			break;
 
-	default:
-		LOG_WRN("Unexpected/unhandled button id: %d", event.button_pin);
+		default:
+			LOG_WRN("Unexpected/unhandled button id: %d", msg.button_pin);
+		}
+		STACK_USAGE_PRINT("button_msg_thread", &button_msg_thread_data);
 	}
 }
 
 /* Handle Bluetooth LE audio events */
-static void le_audio_evt_handler(enum le_audio_evt_type event)
+static void le_audio_msg_thread(void)
 {
 	int ret;
 	uint32_t pres_delay;
+	uint32_t bitrate;
+	uint32_t sampling_rate;
+	const struct zbus_channel *chan;
 
-	LOG_DBG("Received event = %d, current state = %d", event, strm_state);
-	switch (event) {
-	case LE_AUDIO_EVT_STREAMING:
-		LOG_DBG("LE audio evt streaming");
+	while (!zbus_sub_wait(&le_audio_evt_sub, &chan, K_FOREVER)) {
+		struct le_audio_msg msg;
 
-		if (strm_state == STATE_STREAMING) {
-			LOG_DBG("Got streaming event in streaming state");
-			break;
-		}
+		zbus_chan_read(chan, &msg, K_MSEC(100));
```

Comment:
We discussed this and we need to take the suggestion from the documentation into account: 

> Choose the timeout of [zbus_chan_read()](https://docs.zephyrproject.org/latest/services/zbus/index.html#c.zbus_chan_read) after receiving a notification from [zbus_sub_wait()](https://docs.zephyrproject.org/latest/services/zbus/index.html#c.zbus_sub_wait) carefully because the channel will always be unavailable during the VDED execution. Using K_NO_WAIT for reading is highly likely to return a timeout error if there are more than one subscriber. For example, consider the [Fig. 28](https://docs.zephyrproject.org/latest/services/zbus/index.html#zbus-vded) again and notice how T3 and T4's read attempts would definitely fail with K_NO_WAIT. For more details, check the [Virtual Distributed Event Dispatcher](https://docs.zephyrproject.org/latest/services/zbus/index.html#virtual-distributed-event-dispatcher) section.

Furthermore, these are not very time critical operations, so I think 100ms is good here.